### ISSUE TITLE
Web terminal clears its history after some time, only keeps `sudo visudo`

### ISSUE THREAD

**Issue by phil-blain** [cirruslabs/cirrus-ci-web#439]
Hi! Thanks for a great service :)!

I noticed that after using the web terminal for some time ("Re-run with terminal access"), after about 10 minutes or so  the terminal is "reset" and the shell seems to loose its history. Pressing up arrow yields `sudo visudo` and nothing else.

I also think that after some time of inactivity the session is closed. Not huge issues but still I thought I'd report them.

**Comment by phil-blain** [comment]
If it matters, this is on the macOS Catalina image, I haven't tried anything else.

**Comment by phil-blain** [comment]
Interestingly, in the log of the "Wait for terminal" task, [I see](https://cirrus-ci.com/task/6599595017895936?logs=Wait%20for%20terminal#L2)
```
Terminal host failed: rpc error: code = Internal desc = stream terminated by RST_STREAM with error code: INTERNAL_ERROR
```

**Comment by edigaryev** [comment]
Hi Philippe 👋 Thanks for reporting this!

I've investigated the issue and found out that we've had all gRPC keepalives enabled, except the ones used for [gRPC-Web + websocket mechanism](https://github.com/improbable-eng/grpc-web/blob/master/client/grpc-web/docs/transport.md#socket-based-transports).

I've addressed this in https://github.com/cirruslabs/terminal/pull/25, the fix is now deployed, the PINGs are coming and it seems that the idling for more than a couple of minutes doesn't result in session termination as it was before 🙌

As for the `INTERNAL_ERROR`, I'll keep this ticket open until the underlying issue will be nailed too.
